internal/services/manager.go

```diff
@@ -49,46 +153,78 @@ func StartProcessManager(
 	if processComposeBackground {
 		flags = append(flags, "-t=false")
 		cmd := exec.Command(processComposePath, flags...)
-		return runProcessManagerInBackground(cmd, processComposePidfile, processComposeLogfile)
+		return runProcessManagerInBackground(cmd, port)
 	}
 
-	return runProcessManagerInForeground(processComposePath, flags, processComposePidfile)
+	cmd := exec.Command(processComposePath, flags...)
+	return runProcessManagerInForeground(cmd, port, w)
 }
 
-func runProcessManagerInForeground(processComposePath string, flags []string, processComposePidfile string) error {
-	cmd := exec.Command(processComposePath, flags...)
+func runProcessManagerInForeground(cmd *exec.Cmd, port int, w io.Writer) error {
+	runID := uuid.New().String()
 
 	if err := cmd.Start(); err != nil {
 		return fmt.Errorf("failed to start process-compose: %w", err)
 	}
 
-	if err := os.WriteFile(processComposePidfile, []byte(strconv.Itoa(cmd.Process.Pid)), 0666); err != nil {
-		return fmt.Errorf("failed to write pidfile: %w", err)
+	globalProcessComposeConfig := readGlobalProcessComposeConfig()
+
+	projectConfig := projectConfig{
+		Pid:  cmd.Process.Pid,
+		Port: port,
+	}
+
+	globalProcessComposeConfig.Instances[runID] = projectConfig
+
+	if err := os.WriteFile(processComposeUIDFile, []byte(runID), 0666); err != nil {
```

Comment:
I'm not sure this is the best idea. Reasons:

1. The `.devbox` folder's semantics are currently that it can be blown away, and the devbox CLI will just re-create it, and one can continue.
2. This `.devbox/compose.uid` file changes those semantics. And I don't think its worth it for the goal of "move resiliency".

Suggestion:
Could we instead derive the ID as `hash(projectDir)`? This will not be resilient to moves, but I think we can document that services need to be stopped prior to relocating projects.

Not very opinionated about the `hash(projectDir)` implementation, but as a suggestion it could be like:
```
		h := fnv.New32a()
 		_, err := h.Write([]byte(projectDir))
 		if err != nil {
 			return "", errors.WithStack(err)
 		}
 		hashed := strconv.FormatUint(uint64(h.Sum32()), 10)
```
from: `plugin/files.go` in https://github.com/jetpack-io/devbox/pull/874/files

Reply:
Updated this: 

1. I switched to using projectDir as the key. In addition to being simpler, its more resilient to situations where process-compose crashes/fails and without cleaning up the config
2. I kept everything in one file. Splitting to multiple files meant I needed code to get all the files in a directory, parse them, etc. Seemed messy
3. Since everything is in one file, I added file locking/unlocking to avoid situations where 2 projects try to write/read the file at the same time. I’m using a `flock` syscall to lock the file between processes, but I’m keeping this best effort and not throwing an error if the lock fails